src/Streamly/Internal/Network/Socket.hs

```diff
@@ -497,6 +498,17 @@ writeStrings encode h =
 fromBytesWithBufferOf :: MonadIO m => Int -> Socket -> SerialT m Word8 -> m ()
 fromBytesWithBufferOf n h m = fromChunks h $ AS.arraysOf n m
 
+-- | Transform a stream of @a@ to @(Just a)@ then intersperse a Nothing into the input 
+-- stream after every n elements then interject Nothing every @t@ seconds.
+-- @since 0.8.0
+{-# INLINE chunksOfTimeout #-}
+chunksOfTimeout :: (MonadIO m , IsStream t, MonadAsync m)
+    => Int -> Double -> t m a -> t m (Maybe a)
+chunksOfTimeout n t =   
```

Comment:
1. This should go in the same module as `chunksOf`/`intervalsOf`. 
2. We could have a corresponding `chunksOf` as well (i.e. without a timeout), but we already have a `chunksOf`. So should we call these `justsOf` and `justsOfTimeout` to indicate contiguous chunks of `Just` values?